test/testcafe/framework/page-objects/DeviceChallengePollPageObject.js

```diff
@@ -24,8 +24,8 @@ export default class DeviceChallengePollViewPageObject extends BasePageObject {
     return this.body.find('iframe');
   }
 
-  getIframeAttributes() {
-    return Selector('#custom-uri-container').attributes;
+  getIframeAttributes(id) {
+    return Selector(id).attributes;
```

Comment:
DOM selectors should remain in PageObject

Reply:
I think you can use the same id for both iframes, so in this case, this PageObject method can remain unchanged.

If you did have a reason to use unique ids, then you should add another method to this PageObject class. Anytime you need to hardcode an id in a test, it should be done in the PageObject rather than the test itself. This way the DOM can be changed overtime, but the test will still pass (ideally)

Reply:
Thank you Jared. I will create a new method with a new hard-coded id then, it will be easier for us to add tests on it.